plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,609 @@
+	"down_after_milliseconds": "integer",
+	"flags":                   "string",
+	"last_hello_message":      "integer",
+	"last_ok_ping_reply":      "integer",
+	"last_ping_reply":         "integer",
+	"last_ping_sent":          "integer",
+	"link_pending_commands":   "integer",
+	"link_refcount":           "integer",
+	"name":                    "string",
+	"voted_leader":            "string",
+	"voted_leader_epoch":      "integer",
+}
+
+// Supported fields for "redis_sentinel_replicas"
+var measurementReplicasFields = map[string]string{
+	"down_after_milliseconds": "integer",
+	"flags":                   "string",
+	"info_refresh":            "integer",
+	"last_ok_ping_reply":      "integer",
+	"last_ping_reply":         "integer",
+	"last_ping_sent":          "integer",
+	"link_pending_commands":   "integer",
+	"link_refcount":           "integer",
+	"master_host":             "string",
+	"master_link_down_time":   "integer",
+	"master_link_status":      "string",
+	"master_port":             "integer",
+	"name":                    "string",
+	"role_reported":           "string",
+	"role_reported_time":      "integer",
+	"slave_priority":          "integer",
+	"slave_repl_offset":       "integer",
+}
+
+func init() {
+	inputs.Add("redis_sentinel", func() telegraf.Input {
+		return &RedisSentinel{}
+	})
+}
+
+func (r *RedisSentinelClient) BaseTags() map[string]string {
+	tags := make(map[string]string)
+	for k, v := range r.tags {
+		tags[k] = v
+	}
+	return tags
+}
+
+func (r *RedisSentinel) SampleConfig() string {
+	return `
+  ## specify servers via a url matching:
+  ##  [protocol://][:password]@address[:port]
+  ##  e.g.
+  ##    tcp://localhost:26379
+  ##    tcp://:password@192.168.99.100
+  ##    unix:///var/run/redis-sentinel.sock
+  ##
+  ## If no servers are specified, then localhost is used as the host.
+  ## If no port is specified, 26379 is used
+  servers = ["tcp://localhost:26379"]
+
+  ## specify server password
+  # password = "s#cr@t%"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = true
+`
+}
+
+func (r *RedisSentinel) Description() string {
+	return "Read metrics from one or many redis-sentinel servers"
+}
+
+func (r *RedisSentinel) Init() error {
+	if len(r.Servers) == 0 {
+		r.Servers = []string{"tcp://localhost:26379"}
+	}
+
+	r.clients = make([]*RedisSentinelClient, len(r.Servers))
+
+	tlsConfig, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	for i, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("unable to parse to address %q: %s", serv, err.Error())
+		}
+
+		if u.Scheme != "tcp" && u.Scheme != "unix" {
+			return fmt.Errorf("invalid scheme %q. expected tcp or unix", u.Scheme)
+		}
+
+		password := ""
+
+		if len(r.Password) > 0 {
+			password = r.Password
+		}
```

Comment:
This can be written as
```suggestion
		password := r.Password
```

Reply:
@srebhan Addressed in https://github.com/influxdata/telegraf/pull/10042/commits/1671e6c713c393652423da68722427ee10e9a2c9